### PR TITLE
Update IDE Public APIs

### PR DESCRIPTION
**Pull request by 333fred**
Rename public API instances of Record to RecordClass, as part of https://github.com/dotnet/roslyn/issues/50297.

### CODE REVIEW

**Comment by 333fred** [comment]
Dammit, windows did some weird stuff with the permissions. Give me a minute to fix that up.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by 333fred** [comment]
Alright, permissions fixed up.

File: src/EditorFeatures/Core.Wpf/Classification/ClassificationTypeFormatDefinitions.cs [hunk 1]
```diff
@@ -187,8 +187,8 @@ public UserTypeClassesFormatDefinition()
         #endregion
         #region User Types - Records
         [Export(typeof(EditorFormatDefinition))]
-        [ClassificationType(ClassificationTypeNames = ClassificationTypeNames.RecordName)]
-        [Name(ClassificationTypeNames.RecordName)]
+        [ClassificationType(ClassificationTypeNames = ClassificationTypeNames.RecordClassName)]
+        [Name(ClassificationTypeNames.RecordClassName)]
```
Comment (JoeRobich) [inline src/EditorFeatures/Core.Wpf/Classification/ClassificationTypeFormatDefinitions.cs:191]: Should the class name also update to `UserTypeRecordClassesFormatDefinition`? I don't belive it is ever surfaced to the user, so likely not an issue.

**Review by JoeRobich — APPROVED** [review]

File: src/EditorFeatures/Core.Wpf/Classification/ClassificationTypeFormatDefinitions.cs (hunk 1 above)
Comment (333fred) [reply src/EditorFeatures/Core.Wpf/Classification/ClassificationTypeFormatDefinitions.cs:191]: Yeah, it's internal, so it's not an issue.

**Pull request merged**